src/zenml/integrations/gcp/model_deployers/vertex_model_deployer.py

```diff
@@ -0,0 +1,270 @@
+        self,
+        id: UUID,
+        config: ServiceConfig,
+        timeout: int = DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+    ) -> BaseService:
+        """Deploy a model to Vertex AI.
+
+        Args:
+            id: the UUID of the service to be created.
+            config: the configuration of the model to be deployed.
+            timeout: the timeout for the deployment operation.
+
+        Returns:
+            The ZenML Vertex AI deployment service object.
+        """
+        with track_handler(AnalyticsEvent.MODEL_DEPLOYED) as analytics_handler:
+            config = cast(VertexAIDeploymentConfig, config)
+            service = self._create_deployment_service(
+                id=id, config=config, timeout=timeout
+            )
+            logger.info(
+                f"Creating a new Vertex AI deployment service: {service}"
+            )
+            service.start(timeout=timeout)
+
+            client = Client()
+            stack = client.active_stack
+            stack_metadata = {
+                component_type.value: component.flavor
+                for component_type, component in stack.components.items()
+            }
+            analytics_handler.metadata = {
+                "store_type": client.zen_store.type.value,
+                **stack_metadata,
+            }
+        return service
+
+    def perform_stop_model(
+        self,
+        service: BaseService,
+        timeout: int = DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+        force: bool = False,
+    ) -> BaseService:
+        """Stop a Vertex AI deployment service.
+
+        Args:
+            service: The service to stop.
+            timeout: Timeout in seconds to wait for the service to stop.
+            force: If True, force the service to stop.
+
+        Returns:
+            The stopped service.
+        """
+        service.stop(timeout=timeout, force=force)
+        return service
+
+    def perform_start_model(
+        self,
+        service: BaseService,
+        timeout: int = DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+    ) -> BaseService:
+        """Start a Vertex AI deployment service.
+
+        Args:
+            service: The service to start.
+            timeout: Timeout in seconds to wait for the service to start.
+
+        Returns:
+            The started service.
+        """
+        service.start(timeout=timeout)
+        return service
+
+    def perform_delete_model(
+        self,
+        service: BaseService,
+        timeout: int = DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+        force: bool = False,
+    ) -> None:
+        """Delete a Vertex AI deployment service.
+
+        Args:
+            service: The service to delete.
+            timeout: Timeout in seconds to wait for the service to stop.
+            force: If True, force the service to stop.
+        """
+        service = cast(VertexDeploymentService, service)
+        service.stop(timeout=timeout, force=force)
+        service.stop()
```

Comment:
Why do we have to stop twice? Also, how does stopping a service actually delete a model?